c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -2467,7 +2493,11 @@ class TranslateAction : public clang::ASTFrontendAction {
     virtual std::unique_ptr<clang::ASTConsumer>
     CreateASTConsumer(clang::CompilerInstance &Compiler,
                       llvm::StringRef InFile) {
-        if(this->getCurrentFileKind().getLanguage() != InputKind::Language::C) {
+#if CLANG_VERSION_MAJOR < 10
+    if(this->getCurrentFileKind().getLanguage() != InputKind::Language::C) {
```

Comment:
You could merge a lot of the two branches using line tricks such that only `InputKind::Language::C` and `Language::C` are under the conditional.